cmake/DaemonFlags.cmake

```diff
@@ -201,256 +184,350 @@ if (MSVC)
-        set_c_cxx_flag("-flto")
-        set_linker_flag("-flto")
-
-        # For LTO compilation we must send a copy of all compile flags to the linker
-        set_linker_flag("${CMAKE_CXX_FLAGS}")
-
-        # Use gcc-ar and gcc-ranlib instead of ar and ranlib so that we can use
-        # slim LTO objects. This requires a recent version of GCC and binutils.
-        if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL GNU)
-            if (USE_SLIM_LTO)
-                string(REGEX MATCH "^([0-9]+.[0-9]+)" _version "${CMAKE_CXX_COMPILER_VERSION}")
-                get_filename_component(COMPILER_BASENAME "${CMAKE_C_COMPILER}" NAME)
-                if (COMPILER_BASENAME MATCHES "^(.+-)g?cc(-[0-9]+\\.[0-9]+\\.[0-9]+)?(\\.exe)?$")
-                    set(TOOLCHAIN_PREFIX ${CMAKE_MATCH_1})
-                endif()
-
-                find_program(GCC_AR
-                    NAMES
-                        "${TOOLCHAIN_PREFIX}gcc-ar"
-                        "${TOOLCHAIN_PREFIX}gcc-ar-${_version}"
-                    DOC "gcc provided wrapper for ar which adds the --plugin option"
-                )
-
-                find_program(GCC_RANLIB
-                    NAMES
-                        "${TOOLCHAIN_PREFIX}gcc-ranlib"
-                        "${TOOLCHAIN_PREFIX}gcc-ranlib-${_version}"
-                    DOC "gcc provided wrapper for ranlib which adds the --plugin option"
-                )
-
-                mark_as_advanced(GCC_AR GCC_RANLIB)
-
-                # Override standard ar and ranlib with the gcc- versions
-                if (GCC_AR)
-                    set(CMAKE_AR ${GCC_AR})
-                endif()
-                if (GCC_RANLIB)
-                    set(CMAKE_RANLIB ${GCC_RANLIB})
-                endif()
-
-                try_c_cxx_flag(NO_FAT_LTO_OBJECTS "-fno-fat-lto-objects")
-            else()
-                try_c_cxx_flag(FAT_LTO_OBJECTS "-ffat-lto-objects")
-            endif()
-        endif()
+        set_c_cxx_flag("/GL" MINSIZEREL)
+        set_c_cxx_flag("/GL" RELWITHDEBINFO)
+        set_c_cxx_flag("/GL" RELEASE)
+        set_linker_flag("/LTCG" MINSIZEREL)
+        set_linker_flag("/LTCG" RELWITHDEBINFO)
+        set_linker_flag("/LTCG" RELEASE)
     endif()
-
+else()
+	# Minimum language standards supported.
+
+	option(USE_RECOMMENDED_C_STANDARD "Use recommended C standard" ON)
+	mark_as_advanced(USE_RECOMMENDED_C_STANDARD)
+
+	if (USE_RECOMMENDED_C_STANDARD)
+		# Required when building gzip or it would complain about
+		# implicitly defined lseek, read, write and close.
+		try_c_flag(GNU89 "-std=gnu89")
+		if (NOT FLAG_GNU89)
+			set(NOT_GNU_C ON)
+			try_c_flag(C99 "-std=c99")
+			if (NOT FLAG_C99)
+				message(FATAL_ERROR "GNU89 or C99 not supported by compiler")
+			endif()
+		endif()
+	endif()
+
+	option(USE_RECOMMENDED_CXX_STANDARD "Use recommended C++ standard" ON)
+	mark_as_advanced(USE_RECOMMENDED_CXX_STANDARD)
+
+	if (USE_RECOMMENDED_CXX_STANDARD)
+		# PNaCl only defines isascii if __STRICT_ANSI__
```

Comment:
Who needs "isascii"? I don't find any references to this in Daemon or Unvanquished `src` or `libs`.

Reply:
It's in the libc++ provided by PNaCl:

```
In file included from Unvanquished/daemon/src/common/Common.h:35:
In file included from Unvanquished/daemon/src/engine/qcommon/q_shared.h:112:
In file included from build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/random:1645:
In file included from build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/istream:163:
In file included from build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/ostream:138:
In file included from build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/ios:216:
build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/__locale:553:16: error: 
      use of undeclared identifier 'isascii'
        return isascii(__c) ? (__tab_[static_cast<int>(__c)] & __m) !=0 : false;
               ^
build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/__locale:577:19: error: 
      use of undeclared identifier 'isascii'
            if (!(isascii(*__low) && (__tab_[static_cast<int>(*__low)] & __m)))
                  ^
build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/__locale:560:22: error: 
      use of undeclared identifier 'isascii'
            *__vec = isascii(*__low) ? __tab_[static_cast<int>(*__low)] : 0;
                     ^
build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/__locale:568:17: error: 
      use of undeclared identifier 'isascii'
            if (isascii(*__low) && (__tab_[static_cast<int>(*__low)] & __m))
                ^
build/deps/linux-amd64-default_10/pnacl/le32-nacl/include/c++/v1/__locale:577:19: error: 
      use of undeclared identifier 'isascii'
            if (!(isascii(*__low) && (__tab_[static_cast<int>(*__low)] & __m)))
                  ^
```